sockeye/inference.py

```diff
@@ -862,17 +897,27 @@ def _get_inference_input(self,
         for j, trans_input in enumerate(trans_inputs):
             num_tokens = len(trans_input)  # includes eos
             max_output_lengths.append(self._get_max_output_length(num_tokens))
-            source[j, :num_tokens, 0] = tokens2ids(trans_input.tokens, self.source_vocabs[0])
+            num_tokens = num_tokens if not trans_input.source_prefix else num_tokens + len(trans_input.source_prefix.split())
+            if not trans_input.source_prefix:
+                source[j, :num_tokens, 0] = tokens2ids(trans_input.tokens, self.source_vocabs[0])
+            else:
+                #adding source prefix to every chunk
+                source[j, :num_tokens, 0] = tokens2ids(trans_input.source_prefix.split() + \
```

Comment:
maybe we can also add a property to `TranslatorInput` that returns the `source_prefix` or an empty list if the field is `None`, then you can do this in a one-liner without the if:
` source[j, :num_tokens, 0] = tokens2ids(trans_input.source_prefix + trans_input.tokens, self.source_vocabs[0])`

Reply:
Yeah I followed your suggestion in the revision version.